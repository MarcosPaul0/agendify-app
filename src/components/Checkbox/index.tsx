import { COLORS } from '@constants/colors.constant';
import { Check } from 'phosphor-react-native';
import { View } from 'react-native';
import { ICheckboxProps } from './interfaces/checkboxProps.interface';

export function Checkbox({ isChecked, isRounded = false }: ICheckboxProps) {
  const checkedBackground = isChecked ? 'bg-BLUE_500' : 'bg-GRAY_100';
  const round = isRounded ? 'rounded-full' : 'rounded-md';

  return (
    <View
      className={`
      border-2 border-BLUE_500
      ${round}
      ${checkedBackground} h-6 w-6
    `}
    >
      {isChecked && <Check size={20} color={COLORS.GRAY_50} />}
    </View>
  );
}
