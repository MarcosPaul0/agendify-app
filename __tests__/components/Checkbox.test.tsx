import { render } from '@testing-library/react-native';
import { Checkbox } from '@components/Checkbox';

describe('Checkbox', () => {
  it('Match with snapshot', async () => {
    const tree = render(<Checkbox isChecked />);

    expect(tree).toMatchSnapshot();
  });
});
