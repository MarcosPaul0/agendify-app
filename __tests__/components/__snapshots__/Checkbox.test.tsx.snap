// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Match with snapshot 1`] = `
<View
  style={
    [
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      {
        "borderBottomColor": "#3763A4",
        "borderLeftColor": "#3763A4",
        "borderRightColor": "#3763A4",
        "borderTopColor": "#3763A4",
      },
      {
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
      },
      {
        "backgroundColor": "#3763A4",
      },
      {
        "height": 24,
      },
      {
        "width": 24,
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="20"
    bbWidth="20"
    color="#F8F9FA"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    size={20}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          undefined,
          undefined,
          {},
        ],
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    tintColor="#F8F9FA"
    vbHeight={256}
    vbWidth={256}
    width={20}
  >
    <RNSVGGroup>
      <RNSVGRect
        fill={null}
        height="256"
        propList={
          [
            "fill",
          ]
        }
        width="256"
        x="0"
        y="0"
      />
      <RNSVGPath
        d="M216 72 104 184 48 128"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4294507002,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="16"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
